### PR TITLE
Core/GameObjects: Use serverOnly property of GameObjectTemplate for a…

### PR DESCRIPTION
**Pull request by Kitzunu**
…ll that have it

* cherry-pick commit (https://github.com/TrinityCore/TrinityCore/commit/4ee64a232ad830255bc6779e595d412c934b6b74)

## Changes Proposed:
-  All gobs that use serverSide property should be invisible
-  

## Issues Addressed:

- Closes 

## Tests Performed:

- 
- 

## How to Test the Changes:

1.
2.
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/game/Entities/GameObject/GameObject.cpp [hunk 1]
```diff
@@ -1252,7 +1252,7 @@ bool GameObject::IsNeverVisible() const
     if (WorldObject::IsNeverVisible())
         return true;
 
-    if (GetGoType() == GAMEOBJECT_TYPE_SPELL_FOCUS && GetGOInfo()->spellFocus.serverOnly == 1)
+    if (GetGoType()->GetServerOnly())
```
Comment (Kitzunu) [inline src/server/game/Entities/GameObject/GameObject.cpp:1255]:
```suggestion
    if (GetGOInfo()->GetServerOnly())
```

**Comment by LeFroid** [comment]
There is an open issue in TC referring to this change: https://github.com/TrinityCore/TrinityCore/issues/28825 
Possible to do regression test on the same?

**Comment by Kitzunu** [comment]
I will look into it

**Comment by Kitzunu** [comment]
https://github.com/TrinityCore/TrinityCore/commit/5d5ca30b232586ad378671bf0ac8407495dea188 will fix it. I will work on it later

**Comment by Kofmy** [comment]
degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates

**Pull request closed without merging**